arangod/Replication2/StateMachines/Document/DocumentStateShardHandler.cpp

```diff
@@ -226,32 +226,31 @@ auto DocumentStateShardHandler::prepareShardsForLogReplay() noexcept -> void {
     // LocalDocumentID.
     for (auto const& index : shard->getPhysical()->getReadyIndexes()) {
       if (index->type() == Index::TRI_IDX_TYPE_INVERTED_INDEX) {
-        basics::downCast<iresearch::IResearchRocksDBInvertedIndex>(*index)
-            .commit(true);
+        auto& idx =
+            basics::downCast<iresearch::IResearchRocksDBInvertedIndex>(*index);
+        TRI_ASSERT(!idx._isCreation) << "Inverted index still in creation mode";
+        auto res = idx.commit(true);
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+        TRI_ASSERT(res.ok()) << "Failed to do first Inverted index commit";
+        res = idx.commit(true);
```

Comment:
Is it actually intentional that we do two commits here in maintainer mode? can that have any side effects?

Reply:
Yep, we do second commit, and check that nothing changes